packaging/rpm/flightctl-agent.spec

```diff
@@ -2,33 +2,50 @@
 
 Name: flightctl-agent
 Version: 0.0.1
-Release: 1%{?dist}
+Release: 2%{?dist}
 Summary: Flightctl Agent
 
 License: XXX
 URL: https://github.com/flightctl/flightctl
-Source0: flightctl-agent-0.0.1.tar
+Source0: flightctl-agent-0.0.1.tar.gz
+
+BuildRequires: golang
+BuildRequires: make
+BuildRequires: git
+BuildRequires: openssl-devel
+Requires: openssl
 
 %description
 Flightctl Agent is a component of the flightctl tool.
 
 %prep
-%setup -q
+%setup -q -n flightctl-agent-0.0.1
 
 %build
 
+# if this is a buggy version of go we need to set GOPROXY as workaround
+# see https://github.com/golang/go/issues/61928
+GOENVFILE=$(go env GOROOT)/go.env
+if [[ ! -f "{$GOENVFILE}" ]]; then
+    export GOPROXY='https://proxy.golang.org,direct'
+fi
+
+make build
```

Comment:
is this a macro from Packit or something like that? because you have removed the build target from the Makefile

Reply:
I didn't remove the build target, it's still there :-D

https://github.com/flightctl/flightctl/pull/69/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R59

What I did was removing the dependency from the rpm call. Instead what we do, to be able to us builders from source code, is calling make build inside the rpm spec, so the srpm is packaged with the whole sourcecode (not the binaries), and the .spec file... then the builders will simply compile the go for each architecture.

Reply:
ohhh true! hahah, I saw a red line on build, and boom! my bad... I need more coffee...